packages/discord.js/src/structures/Webhook.js

```diff
@@ -211,7 +208,7 @@ class Webhook {
 
     const { body, files } = await messagePayload.resolveFiles();
     const d = await this.client.rest.post(Routes.webhook(this.id, this.token), { body, files, query, auth: false });
-    return this.client.channels?.cache.get(d.channel_id)?.messages._add(d, false) ?? new (getMessage())(this.client, d);
```

Comment:
instead of always returning `d` could you try treating both cases of optional chaining differently? If the client doesn't have a channels property it's a WebhookClient -> return d, if it has a ChannelManager but the channel isn't cached construct a Message. That way interactions wouldn't have APIMessages (what the initial pr intended) while keeping WebhookClients as before